test/Sentry.Tests/SentryPropagationContextTests.cs

```diff
@@ -2,6 +2,43 @@ namespace Sentry.Tests;
 
 public class SentryPropagationContextTests
 {
+    [Fact]
+    public void CopyConstructor_CreatesCopy()
+    {
+        var original = new SentryPropagationContext();
+        original.GetOrCreateDynamicSamplingContext(new SentryOptions {Dsn = ValidDsn});
+
+        var copy = new SentryPropagationContext(original);
+
+        Assert.Equal(original.TraceId, copy.TraceId);
+        Assert.Equal(original.SpanId, copy.SpanId);
+        Assert.Equal(original._dynamicSamplingContext, copy._dynamicSamplingContext);
+    }
+
+    [Fact]
+    public void GetOrCreateDynamicSamplingContext_DynamicSamplingContextIsNull_CreatesDynamicSamplingContext()
+    {
+        var options = new SentryOptions { Dsn = ValidDsn };
+        var propagationContext = new SentryPropagationContext();
+
+        Assert.Null(propagationContext._dynamicSamplingContext); // Sanity check
+        _ = propagationContext.GetOrCreateDynamicSamplingContext(options);
+
+        Assert.NotNull(propagationContext._dynamicSamplingContext);
+    }
+
+    [Fact]
+    public void GetOrCreateDynamicSamplingContext_DynamicSamplingContextIsNotNull_ReturnsSameDynamicSamplingContext()
+    {
+        var options = new SentryOptions { Dsn = ValidDsn };
+        var propagationContext = new SentryPropagationContext();
+        var firstDynamicSamplingContext = propagationContext.GetOrCreateDynamicSamplingContext(options);
+
+        var secondDynamicSamplingContext = propagationContext.GetOrCreateDynamicSamplingContext(options);
+
+        Assert.Same(firstDynamicSamplingContext, secondDynamicSamplingContext);
+    }
+
     [Fact]
     public void CreateFromHeaders_TraceHeaderNullButBaggageExists_CreatesPropagationContextWithoutDynamicSamplingContext()
```

Comment:
I think we need another test where the traceheader is null and there is no baggage, to make explicit what our expectations are in that scenario.